[T13-3] JxMusic

handshou opened the pull request:
@hidingmode @HaydenPhillips @NancyQuris @Gongjie663746 

JxMusic v1.1 changelog:
- Add `play` command to play 1 hardcoded track in the `/library` folder
- Add `pause` command to pause the track
- Add `stop` command to stop the track
The 3 commands are only working on Windows for now, not working on Mac OS. We are planning to resolve the issue by v1.2.

parvathy-sudhir-pillai commented on docs/AboutUs.adoc:
```diff
@@ -4,7 +4,7 @@
 :stylesDir: stylesheets
 
-AddressBook - Level 4 was developed by the https://se-edu.github.io/docs/Team.html[se-edu] team. +
+JxMusic(AddressBook - Level 4) was developed by the CS2103-Ay1819S1-T13-3 developer (are we supposed to add links) (https://se-edu.github.io/docs/Team.html[se-edu]) team. +
```
What links? Add your portfolio and GitHub links only for now

parvathy-sudhir-pillai commented on docs/DeveloperGuide.adoc:
```diff
@@ -835,54 +835,323 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+1.  User requests to list all the playlists.
+2.  System displays names of all playlists in the library.
++
+Use case ends.
```
No extensions? What if there are no playlists?

parvathy-sudhir-pillai commented on docs/DeveloperGuide.adoc:
```diff
@@ -835,54 +835,323 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+*Extensions*
+
+[none]
+* 4a. There is no list matches the name.
```
No list matches the name.

parvathy-sudhir-pillai reviewed (commented):
*Overall
Good start, Team. However, some areas need more work. Hope you can tackle those areas in upcoming features.

You seem to have a lot of features. Make sure that you scope out your work properly for each version.
You can consider skipping some of the less important features or push to v2.0.

Some detailed feedback:
* ReadMe adoc
1. Homepage still has AB-4 as the product name.
2. Travis build status badge should point to the build status of your own repo, not se-edu/AB-4. 
Other badges should be updated to match their project or removed.
3. Update product description. Still refers to AB-4
4. AB-4 should be acknowledged in the Acknowledgements section

* Contact Us adoc
1. Update details and links

* About Us adoc
1. Roles and components must be assigned to members

* UserGuide adoc
1. Update By:  Since:  at the beginning of the page
2. Update Introduction and Quick Start sections to reflect your product's

* DeveloperGuide adoc
1. Update Product Scope and value proposition to be your product's
2. Grammar check on Use cases
3. Update NFR and Glossary

parvathy-sudhir-pillai reviewed (commented):
### Feedback for v1.2

* Can update the GitHub link to your product webpage instead of AB-4:
![t13-1](https://user-images.githubusercontent.com/9066342/47263529-ecf0ff00-d535-11e8-8b53-3d08c6c57fc2.png)

**Contact Us**
* Update issue tracker link

**User guide**
* Update commands/descriptions that refer to AB-4 rather than your app's. For e.g.
![t13-2](https://user-images.githubusercontent.com/9066342/47263593-27a76700-d537-11e8-9a28-279868435d58.png)
* Examples should be realistic
![t13-3](https://user-images.githubusercontent.com/9066342/47263627-7ead3c00-d537-11e8-92f8-04a5628def3a.png)

**Developer guide**
* No significant updates from previous milestone
* Not enough new/updated design descriptions and UML diagrams to give feedback on. Please try to add more soon.
* Keep in mind that we discourage low level diagrams that span multiple components. Try to draw diagrams at multiple levels: one that shows component level interactions and others shown low level interactions within one components.

### Overall
Work on making the product user-testable soon, and catch up on documentation work. The progress at the moment, at least w.r.t documentation seems a bit behind what is required at this stage 😟

davidchoo12 committed:
Merge pull request #58 from CS2103-AY1819S1-T13-3/gongjie

add list tracks command

davidchoo12 committed:
Fix LibraryParser to take command phrase

davidchoo12 committed:
Merge branch 'v1.3' of https://github.com/CS2103-AY1819S1-T13-3/main into v1.3

davidchoo12 committed:
Add space for PlaylistList and TrackList commands

As LibraryParser is fixed, the PlaylistList and TrackList commands
can use multiple words command phrase.
Add the commands' cases to LibraryParserTest
Fix LogicManagerTest to use PlaylistListCommand instead of ListCommand.

davidchoo12 committed:
Update architecture diagram

Includes Player component

davidchoo12 committed:
Refactor JxMusicPlayer to PlayerManager

davidchoo12 committed:
Refactor PlayPlaylistCommand to PlayCommand

Gongjie663746 committed:
update StorageClassDiagram.png

NancyQuris committed:
update issue tracker link

NancyQuris committed:
Merge branch 'v1.3' into ui

handshou committed:
Add TrackAddCommand

Update gitignore for vscode

NancyQuris committed:
Merge branch 'v1.3' into UI

HaydenPhillips committed:
Merge remote-tracking branch 'origin/v1.3' into hayden

HaydenPhillips committed:
Add stacktrace flag for debugging

NancyQuris committed:
update ObservableTrackList to match UI component

HaydenPhillips committed:
Fix build styling errors

Gongjie663746 committed:
add tracksearchcommand

Gongjie663746 committed:
Merge branch 'UI' into tracksearch

handshou committed:
Correct for checkstyle

Gongjie663746 committed:
add tracksearchcommand

handshou committed:
Correct for checkstyle

handshou committed:
Correct for checkstyle

HaydenPhillips committed:
Add PlaylistSearchCommandTest
Add PlaylistSearchCommandParserTest
Refactor DeleteCommandTest to PlaylistDelCommandTest
Update multiple test files

HaydenPhillips committed:
Add PlaylistSearchCommand
Refactor FindCommandParser to PlaylistSearchCommandParser
Refactor DeleteCommand to PlaylistDelCommand
Update FindCommand & LibraryParser

HaydenPhillips committed:
Update CheckStyle

NancyQuris committed:
fix UI null pointer exception

NancyQuris committed:
fix TrackListPanel.java and TrackCard.fxml error

davidchoo12 committed:
Minor strings fix for PlaylistNewCommand and Playlist

davidchoo12 committed:
Refactor PlayerManager to be a singleton

As Player will only be used in Command subclasses,
I converted the PlayerManager to be a singleton.
The singleton is only used in the Command abstract class.
This way, all subclasses will have access to the player instance.

An alternative would be to add a Player parameter to
every Command subclasses similar to how CommandHistory
used to be a parameter before the refactoring stage.
The argument for not going with this alternative is that
not all Command subclass need to use the Player,
just like CommandHistory was only used by undo and redo.
Also, I prefer to avoid further mass refactoring nightmare.

Of course, we need to be mindful of the consequences of
using singleton as singleton class is essentially a global variable.
We try to use it sparingly.
Pretty sure this is the only time where we need singleton.

Honestly, the initial codes of the AddressBook was not very
intuitive/friendly for integrating new components,
or dependency injection.

davidchoo12 committed:
Implement PlayCommand

Add PlayCommandParser

Change LibraryParser switch case for play command
to use PlayCommandParser

davidchoo12 committed:
Remove album art from unplayable mp3 files

Due to the Java bug where JavaFX MediaPlayer
does not play photoshopped album art
https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-8210828

davidchoo12 committed:
Fix command history not showing

After you entered a command in the UI CommandBox,
you should be able to press up/down key to navigate the commands
you have entered.

This is a regression from the refactoring stage.

davidchoo12 committed:
Fix failing tests

SelectCommandSystemTest
As LibraryParser regex is now detecting first 2 alphabetical words
as the command phrase, the test case should expect
unknown command message.

CommandBoxTest
As we are not using ListCommand,
change the command that succeeds to PlaylistListCommand.

NancyQuris committed:
Merge pull request #63 from CS2103-AY1819S1-T13-3/david

Implement PlayCommand

NancyQuris committed:
Merge branch 'v1.3' into UI

NancyQuris committed:
fix coding style error

NancyQuris committed:
fix coding style error

HaydenPhillips committed:
Update CheckStyle

HaydenPhillips committed:
update checkstyle 3.0 ffs

HaydenPhillips committed:
update checkstyle 4.0

NancyQuris committed:
refactor TypicalPlaylists to TypicalPlaylistList; add test file for TrackCard and TrackListPanel

Gongjie663746 committed:
update tracksearch feature

Gongjie663746 committed:
Merge branch 'UI' into tracksearch

NancyQuris committed:
remove BrowserPanel relevant features in test files

NancyQuris committed:
revise coding style

handshou committed:
Add TrackAddCommand tests

handshou committed:
Merge branch 'v1.3' of github.com:CS2103-AY1819S1-T13-3/main into hansel

NancyQuris committed:
fix a part of errors existed

NancyQuris committed:
fix error in variable declaration, with the help from @hidingmode

NancyQuris committed:
fix error in variable declaration, with the help from @David

NancyQuris committed:
Merge branch 'UI' of https://github.com/CS2103-AY1819S1-T13-3/main into UI

handshou committed:
Implement TrackAddCommand, tests and modified TypicalPlaylists to model library for an added track

handshou committed:
Add .project and .classpath configurations to .gitignore

handshou committed:
Cleanup comments in TrackAddCommand and ParserUtil

handshou committed:
Remove .classpath and .project files generated from Visual Studio Code

handshou committed:
Fix TrackAddCommand message usage string

handshou committed:
Update TrackAddCommand success message

handshou committed:
Fix TrackAddCommand message usage

Gongjie663746 committed:
fix the track search command

Gongjie663746 committed:
update userguide for track search

NancyQuris committed:
fix test errors, with the help from @hidingmode

NancyQuris committed:
revise coding style

NancyQuris committed:
comment openWindow test and will fix the error in v1.4

NancyQuris committed:
revise coding style

NancyQuris committed:
create the frame of SeekCommand, SeekCommandParser and add seek command to libraryParser

davidchoo12 committed:
Merge pull request #64 from CS2103-AY1819S1-T13-3/UI

Create TrackListPanel in UI component

NancyQuris committed:
update UserGuide to explain how new mp3 files can be added

davidchoo12 committed:
Add player component into docs

NancyQuris committed:
update UserGuide.adoc

NancyQuris committed:
update some commands' format

Gongjie663746 committed:
add library folder to resources

Gongjie663746 committed:
Merge branch 'UI' into tracksearch

HaydenPhillips committed:
Merge branch 'david' of https://github.com/CS2103-AY1819S1-T13-3/main into david

Gongjie663746 committed:
update trackscanner so that it can copy tracks from resouces folder to library folder

NancyQuris committed:
update sequential diagram of "event-driven nature of design" part in DeveloperGuide.adoc

NancyQuris committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T13-3/main

Gongjie663746 committed:
update tracksearch in userguide doc

HaydenPhillips committed:
Add PlaylistDelCommandParserTest
Add PlaylistDelCommandParser
Add 4 sample Playlists

davidchoo12 committed:
Merge pull request #65 from CS2103-AY1819S1-T13-3/tracksearch

Implement `track search` command and track list panel UI

NancyQuris committed:
update UI image

HaydenPhillips committed:
Merge remote-tracking branch 'origin/v1.3' into hayden

HaydenPhillips committed:
Fix PlaylistSearchCommand usage message

davidchoo12 committed:
Merge pull request #61 from CS2103-AY1819S1-T13-3/hayden

Implement playlist search and delete commands

HaydenPhillips committed:
Update Logic component diagrams in dev guide

davidchoo12 committed:
Fix TrackList to use predicate

davidchoo12 committed:
Ensure /library exists on startup (fix issue #67)

TrackScanner checks if /library doesn't exist or is empty
then it copies the contents of resources/library out

If it is compiled to a jar file, it runs the command line:
unzip "library/*"

Else when running in IDE, it basically copies the contents

davidchoo12 committed:
Bump to v1.3.0

davidchoo12 committed:
Fix log file name

davidchoo12 committed:
Merge branch 'master' into v1.3

handshou committed:
Implement TrackDeleteCommand, TrackDeleteCommandParser, TrackDeleteCommandTest and add Playlist methods: hasTrack(), deleteTrack()

handshou committed:
Merge branch 'v1.3' of github.com:CS2103-AY1819S1-T13-3/main into hansel

handshou committed:
Fix spaces in usage messages and examples

davidchoo12 committed:
Merge pull request #60 from CS2103-AY1819S1-T13-3/hansel

Add/delete track command for managing playlists

handshou committed:
Update model diagram and developerguide.adoc

davidchoo12 committed:
Fix PlaylistDelCommand command phrase

davidchoo12 committed:
Update and fix user guide

Update the quick start
Fix search commands to remove q/
Fix delete commands to remove i/

davidchoo12 committed:
Make search commands work with substrings

Basically use string.contains to search

davidchoo12 committed:
Merge pull request #62 from CS2103-AY1819S1-T13-3/v1.3

V1.3

NancyQuris committed:
update theme color of UI to green, add captain on the top of trackListPanel and playlistListPanel

NancyQuris committed:
update UI image

Gongjie663746 committed:
Merge tag 'v1.3' into gongjie

NancyQuris committed:
Merge branch 'master' into yingnan

NancyQuris committed:
change the structure of the building of filteredTrackList

NancyQuris committed:
update ReadOnlyLibrary interface to match the implementation

handshou committed:
Fix track add and track delete commands

handshou committed:
Implement add multiple tracks support to Track Add Command, add empty playlist message if deleting tracks in empty playlist, and tests

handshou committed:
Add size attribute and isEmpty() to Playlist

handshou committed:
Rework track add, track del and tweak tests. Rename getTypicalLibraryAfterAdd() to getModifiedTypicalLibrary()

handshou committed:
Attempt to fix failing track add test

handshou committed:
Fix checkstyle: unused import

NancyQuris committed:
update Codacy and Coveralls report

NancyQuris committed:
Update README.adoc

NancyQuris committed:
update code quality link

NancyQuris committed:
update Coverage Status

NancyQuris committed:
update Codacy Badge's image

Gongjie663746 committed:
add tests for tracklist command and tracksearch command

NancyQuris committed:
configure codacy to exclude the checking of test files

NancyQuris committed:
Update .codacy.yml

NancyQuris committed:
Update .codacy.yml

NancyQuris committed:
Update .codacy.yml

NancyQuris committed:
Update .codacy.yml

davidchoo12 committed:
Add status for Player and Playable

The status can be used for validation checks when running
player related commands

Eg. if Player is not playing, pause command should announce
there is no playing track to pause

davidchoo12 committed:
Add validation checks for Play, Pause, Stop

davidchoo12 committed:
Add PlayCommandTest

Reorder TypicalPlaylistList to put ANIME as first
so that when playing default playlist, test success

NancyQuris committed:
implement seek command parser and seek command

davidchoo12 committed:
Merge branch 'v1.4' into david

davidchoo12 committed:
Attempt fix failing travis build

Travis fails with "Toolkit already initialized" exception

davidchoo12 committed:
Fix failing travis and appveyor builds

Configure to skip PlayCommandTest when run on travis and appveyor

NancyQuris committed:
Merge branch 'david' into yingnan

NancyQuris committed:
implement seek command

NancyQuris committed:
Merge branch 'UI' into yingnan

NancyQuris committed:
correct exception handling, add track duration to UI

handshou committed:
Implement load tracks in TestApp, fix TrackAddCommand tests, add required methods

NancyQuris committed:
fix getDuration problem

davidchoo12 committed:
Add PlayerManagerTest

NancyQuris committed:
fix some of the bugs in seek command

handshou committed:
Improve TrackAddCommand, TrackDeleteCommand tests, add new TEST Playlist in TypicalPlaylistList

handshou committed:
Fix for codacy

davidchoo12 committed:
Merge pull request #101 from CS2103-AY1819S1-T13-3/david

Add statuses to Player for validation checks in player related commands

davidchoo12 committed:
Merge pull request #68 from CS2103-AY1819S1-T13-3/UI

Update the design of UI

davidchoo12 committed:
Merge pull request #70 from CS2103-AY1819S1-T13-3/hansel

Fix add/delete track command for managing playlists

davidchoo12 committed:
Merge pull request #100 from CS2103-AY1819S1-T13-3/gongjie

Add tests for tracksearch command and tracklist command

NancyQuris committed:
add nancyquirs.adoc

NancyQuris committed:
update nancyquris.adoc

handshou committed:
Merge branch 'v1.4' into hansel

handshou committed:
Merge pull request #105 from CS2103-AY1819S1-T13-3/hansel

TrackAdd TrackDelete into v1.4

davidchoo12 committed:
Make trackset sortable by using TreeSet

davidchoo12 committed:
Rename Ui latest.png to Ui.png (close #103)

handshou committed:
Use equalsIgnoreCase() over equals-toLowerCase()

handshou committed:
Add missing message string by including PREFIX_INDEX

NancyQuris committed:
Revert "update nancyquris.adoc"

This reverts commit eb3e4555399f7a97667b259b20fab6872970dee2.

NancyQuris committed:
fix problem with seek command

NancyQuris committed:
comment test file

NancyQuris committed:
comment test file

NancyQuris committed:
Merge branch 'v1.4' into yingnan

NancyQuris committed:
fix coding style

handshou committed:
Add capability of TrackAdd using indexes

handshou committed:
Add InputType enum for TrackAddCommand constructor

NancyQuris committed:
fix initialisation problem

davidchoo12 committed:
Update user guide

NancyQuris committed:
implement some test cases, comment them to focus later

davidchoo12 committed:
Fix more user guide

davidchoo12 committed:
Remove all "System.out.println"

davidchoo12 committed:
Remove storage main classes for testing

NancyQuris committed:
enable seek function on paused tracks

handshou committed:
Refactor TrackAddCommand and tests

handshou committed:
Change implementation of PREFIX_INDEX and TrackAddCommand

NancyQuris committed:
Merge branch 'v1.4' into yingnan

# Conflicts:
#	docs/UserGuide.adoc

davidchoo12 committed:
Delete and reimplement some tests

davidchoo12 committed:
Fix failing PlayCommandTest

when playing default playlist

handshou committed:
Add tests for adding tracks using PREFIX_INDEX

handshou committed:
Merge branch 'v1.4' into hansel

davidchoo12 committed:
Fix indentation of ExitCommandTest and HelpCommandTest

handshou committed:
Add more TrackAddCommand tests for PREFIX_INDEX

handshou committed:
Merge branch 'hansel' of github.com:CS2103-AY1819S1-T13-3/main into hansel

handshou committed:
Merge branch 'v1.4' of github.com:CS2103-AY1819S1-T13-3/main into hansel

NancyQuris committed:
add seek command test and seek command parser test

NancyQuris committed:
Merge branch 'v1.4' into yingnan

NancyQuris committed:
attempt to fix failing test

NancyQuris committed:
fix coding style erorr

NancyQuris committed:
fix coding style

NancyQuris committed:
update sequential diagram for playlist del command

NancyQuris committed:
change the implementation of duration to pass the tests

NancyQuris committed:
fix coding style error

NancyQuris committed:
add specification to seek command

HaydenPhillips committed:
Delete findCommand
Delete deleteCommandParser
Delete editCommand
Delete editCommand Parser

HaydenPhillips committed:
Update developerDocs
- Update the logicComponentSequenceDiagram.png
- Update the logicComponentSequenceDiagram.pptx
- Correct the playlist del command occurences

davidchoo12 committed:
Add DurationCommand

davidchoo12 committed:
Merge branch 'v1.4' into yingnan

davidchoo12 committed:
Change trackSet to use TreeSet

davidchoo12 committed:
Format duration string

davidchoo12 committed:
Fix codacy failing build

Remove NullPointerException test in PlayableTrack

NancyQuris committed:
Merge pull request #106 from CS2103-AY1819S1-T13-3/yingnan

implement SeekCommand

NancyQuris committed:
update Ui.png

HaydenPhillips committed:
Update adoc: Section 3 - Implementation
Added Playlist Management feature, including all commands with emphasis on playlistNewCommand
Add PlaylistNewActivityDiagram.png
Delete UndoRedo Activity Diagram and image

HaydenPhillips committed:
Merge branch 'v1.4' of https://github.com/CS2103-AY1819S1-T13-3/main into v1.4

parvathy-sudhir-pillai reviewed (commented):
### Mid v1.4 Review
* Developer Guide: Can remove description and SD for undo feature.
* Draw sequence diagrams for your respective features. Your final submission requires the behaviour description and diagrams (either sequence or activity).
* CS2103 encourages multi-level sequence diagrams. Higher level interactions between components (e.g. Logic and Model) in a separate diagram and lower level interactions between sub-components (e.g. Parser and Command) within a component (Logic) in a separate diagram. The lower level diagram should not include interactions between components (e.g. Logic and Model).

**Good job, Team!
All the best for your v1.4 submission and presentation!**

Gongjie663746 committed:
Update dev guide

NancyQuris committed:
update user guide; add images needed for developer guide

davidchoo12 committed:
Update dev guide with Player component and Play command

davidchoo12 committed:
Merge pull request #107 from CS2103-AY1819S1-T13-3/hansel

TrackAddCommand now works with index

NancyQuris committed:
update developer guide

davidchoo12 committed:
Fix failing build from draw io xml

HaydenPhillips committed:
Update UserGuide Introduction

HaydenPhillips committed:
Delete unused systemtests

Gongjie663746 committed:
update PPP

Gongjie663746 committed:
Merge branch 'v1.4' of github.com:CS2103-AY1819S1-T13-3/main into v1.4

Gongjie663746 committed:
update ppp

handshou committed:
Remove addressbook-related appendix

NancyQuris committed:
update PPP

NancyQuris committed:
Update nancyquris.adoc

NancyQuris committed:
add link to yingnan's portfolio

Gongjie663746 committed:
update PPP and about us

Gongjie663746 committed:
update about us

Gongjie663746 committed:
update PPP

Gongjie663746 committed:
delete commented lines

davidchoo12 committed:
Add instructions for manual testing in dev guide

davidchoo12 committed:
Merge branch 'master' into v1.4

NancyQuris committed:
remove irrelevant codes

NancyQuris committed:
delete unused test data files

Gongjie663746 committed:
fix the userguide

NancyQuris committed:
fix errors of tags appeared in PPP

NancyQuris committed:
revise the usage of track del command

NancyQuris committed:
correct indentation level

NancyQuris committed:
update PPP

NancyQuris committed:
Update README.adoc

NancyQuris committed:
Update UserGuide.adoc

NancyQuris committed:
update the name of the project

handshou committed:
Add handshou.adoc, add track add remove feature

handshou committed:
DevGuide: remove one type of delete tracks command

HaydenPhillips committed:
Add Hayden's PPP
Edit DevGuide tags
Edit UserGuide tags

HaydenPhillips committed:
Edit PPP adoc name

davidchoo12 committed:
Add david's PPP

As per requirement of v1.4: https://nus-cs2103-ay1819s1.github.io/cs2103-website/admin/project-w13-v14.html

HaydenPhillips committed:
edit hayden ppp
edit devGuide

handshou committed:
Update personal adoc and devguide

HaydenPhillips committed:
PPP touch-ups Hayden

HaydenPhillips committed:
PPP final Hayden

HaydenPhillips committed:
PPP final Hayden

NancyQuris committed:
update PPP

NancyQuris committed:
update PPP

NancyQuris committed:
update PPP's tag

davidchoo12 committed:
Update david's PPP

Previous PPP exceeds limit of 10 pages
Shrink my diagrams in dev guide

davidchoo12 committed:
Remove conflict markers in user guide

davidchoo12 committed:
Bump to v1.4

davidchoo12 committed:
Update AboutUs on david's part

And Hayden's url

handshou committed:
Update personal adoc, tags in dev and user guide

handshou committed:
Fix checkstyle

handshou committed:
small mistake fix

davidchoo12 committed:
Merge pull request #104 from CS2103-AY1819S1-T13-3/v1.4

V1.4

davidchoo12 committed:
Fix user guide missing new line

handshou committed:
Removing id
